core/src/main/java/org/apache/carbondata/core/util/ThreadLocalTaskInfo.java

```diff
@@ -30,7 +32,7 @@ public static void setCarbonTaskInfo(CarbonTaskInfo carbonTaskInfo) {
   public static CarbonTaskInfo getCarbonTaskInfo() {
     if (null == threadLocal.get()) {
       CarbonTaskInfo carbonTaskInfo = new CarbonTaskInfo();
-      carbonTaskInfo.setTaskId(System.nanoTime());
+      carbonTaskInfo.setTaskId(UUID.randomUUID().toString());
```

Comment:
Please move the generation of UUID to some UUID to some common utility so all methods can use same. In future if we need to change we can change at one place